Integrate nrfx jacob

jacobrosenthal opened the pull request.

jacobrosenthal committed:
nrfx fixes reported upstream

jacobrosenthal committed:
nrf51 adc setup

jacobrosenthal committed:
use ADC_0 instead of ADC

mlaz reviewed (commented):
Shouldn't we do this on the bsp's syscfg?

mlaz committed:
Update syscfg.yml

ADC and other peripheral drivers related macros are defined on bsp/syscfg. Just changing this so it doesn't break. Let's discuss where these definitions should live.
